App.js

```diff
@@ -61,9 +60,58 @@ const AppContainer = createAppContainer(TabNavigator);
 
 export default class App extends Component<Props> {
 
+  events = ["event 2", "event 4", "event 5"];
+
+  async checkPermissions() {
+    const enabled = await firebase.messaging().hasPermission();
+    if (enabled) {
+      console.log("Permissions Enabled");
+    } else {
+      console.log("Asking for permissions");
+      this.askPermissions();
+    }
+  }
+
+  async askPermissions() {
+    try {
+      await firebase.messaging().requestPermission();
+      console.log("Permissions enabled");
+    } catch (error) {
+      console.log("User rejected permissions");
+    }
+  }
+
+  componentDidMount() {
+    this.removeNotificationListener = firebase.notifications().onNotification((notification: Notification) => {
+        // Process your notification as required
+        const { title, body } = notification
+        console.log(this.events.some((event) => {
+          notification.data.tags.includes(event);
+        }));
+
+        if (this.events.some((event) => {
+          if (notification.data.tags.includes(event)) {
+            return true;
+          }
+        })) {
+          this.showAlert(title, body);
+        }
+    });
+  }
+
+  componentWillUnmount() {
+    this.removeNotificationListener();
+  }
+
+  showAlert(title, body) {
+    Alert.alert(
```

Comment:
Can we do any more linking - i.e. open to a specific context?

Reply:
So right now, I'm just sending an alert. I can set it so that when you click a button, it'll take you to that screen